### ISSUE TITLE
Intersection types trigger error

### ISSUE THREAD

**Issue by simPod** [slevomat/coding-standard#228]
> @return annotation of method getNetworks() does not specify type hint for items of its traversable return value. (SlevomatCodingStandard.TypeHints.TypeHintDeclaration.MissingTraversableReturnTypeHintSpecification)

```php
/**
* @return Network[]&Collection
*/
public function getNetworks() : Collection
{
    return new ReadOnlyCollectionWrapper($this->networks);
}
```

This does not work `Network[]&Collection` and triggers error above. This `Collection&Network[]` does not trigger the error.

Am I doing something wrong or is that a bug?

**Comment by ondrejmirtes** [comment]
I think you're misusing intersection types. As PHPStan interprets them and as Slevomat CS expects this information, you should use `Collection|Network[]` which means `Collection of Network`.

**Comment by simPod** [comment]
Well, ok then. It's a bit confusing TBH. Thanks

Anyway, is it still bug or the error was trigger by misuse of intersection types? If the second, we can close.

**Comment by kukulich** [comment]
If you use |, it will work.

**Comment by ondrejmirtes** [comment]
Slevomat CS expects item type specification in a union, it does not know
about intersection types.

**Comment by Majkl578** [comment]
Would be awesome to expand support here as well, otherwise noone can really use PHPStan's advanced annotations (&, parenthesis etc.) with Slevomat CS. 😇

**Comment by simPod** [comment]
@kukulich I know it will. It just doesn't smell right if you know what I mean ;)

**Comment by Majkl578** [comment]
Can you please reopen this as a feature request and reconsider intersection support? It would be just another great step helping PHPStan's adoption.
